### PR TITLE
[PM-289] Add taxIdType for more countries in TaxInfo

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

When adding tax information to a payment method for certain countries, the VAT/GST Tax ID would fail to save. This PR addresses this by adding the remaining `taxIdType` values for all possible countries supported by Stripe.

## Code changes

* **TaxInfo.cs:** Updated the getter for TaxIdType to include all possible tax ID type for every country that stripe supports

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Models/Business/TaxInfo.cs [hunk 1]
```diff
@@ -37,12 +37,24 @@ public string TaxIdType
 
             switch (BillingAddressCountry)
```
Comment (cyprain-okeke) [inline src/Core/Models/Business/TaxInfo.cs:38]: It looks Good. 

If this switch grows or changes frequently, you might consider refactoring it. Using a Dictionary<string, string> for mapping country codes to tax ID types can be more efficient and easier to maintain.

```
private static readonly Dictionary<string, string> TaxIdTypeMap = new Dictionary<string, string>
{
    { "AD", "ad_nrt" },
    { "AE", "ae_trn" },
    { "AF", "af_tin" }, 
    { "AR", "ar_cuit" },
    // ... add all other mappings ...
    { "ZA", "za_vat" },
};
```

Comment (cyprain-okeke) [inline src/Core/Models/Business/TaxInfo.cs:38]: It's also important to handle potential issues with input data. If BillingAddressCountry could be in lowercase, ensure it's transformed to uppercase before the lookup.

File: src/Core/Models/Business/TaxInfo.cs [hunk 2]
```diff
@@ -37,12 +37,24 @@ public string TaxIdType
...
             {
+                case "AD":
+                    _taxIdType = "ad_nrt";
+                    break;
                 case "AE":
```
Comment (cyprain-okeke) [inline src/Core/Models/Business/TaxInfo.cs:43]: There's a repeated value for the country code "AF" - it has the tax ID type "ae_trn", which appears to be the same as "AE". Please, ensure this isn't a mistake.

File: src/Core/Models/Business/TaxInfo.cs (hunk 1 above)
Comment (cturnbull-bitwarden) [reply src/Core/Models/Business/TaxInfo.cs:38]: Nice, thanks for these suggestions. I'll push up a change to address them

**Review by cyprain-okeke — APPROVED** [review]
Look Good

**Pull request merged**